components/tikv_util/src/config.rs

```diff
@@ -142,35 +142,42 @@ impl FromStr for ReadableSize {
             return Err(format!("ASCII string is expected, but got {:?}", s));
         }
 
-        let mut chrs = size_str.chars();
-        let mut number_str = size_str;
-        let mut unit_char = chrs.next_back().unwrap();
-        if unit_char < '0' || unit_char > '9' {
-            number_str = chrs.as_str();
-            if unit_char == 'B' {
-                let b = match chrs.next_back() {
-                    Some(b) => b,
-                    None => return Err(format!("numeric value is expected: {:?}", s)),
-                };
-                if b < '0' || b > '9' {
-                    number_str = chrs.as_str();
-                    unit_char = b;
-                }
-            }
-        } else {
-            unit_char = 'B';
+        // size: digits and '.' as decimal separator
+        let size = size_str
+            .to_string()
+            .chars()
+            .take_while(|c| char::is_ascii_digit(c) || *c == '.')
+            .collect::<String>();
+
+        // unit: alphabetic characters, i.e., no more digits etc expected
+        let (_, unit) = size_str.split_at(size.len());
+
+        let is_unit_invalid = unit
+            .to_string()
+            .trim()
+            .chars()
+            .find(|c| !char::is_ascii_alphabetic(c));
```

Comment:
The check here is redundant because `unit` will be strictly matched to the list anyway.

Reply:
True, I'll fix that.